### PR TITLE
Use Extensions.Identity.Core instead of AspNetCore.Identity

### PR DESCRIPTION
**Pull request by nphmuller**
In AspNetCore 2.0 the Identity project has been split into 2 parts: AspNetCore.Identity and Extensions.Identity.Core.

ExtensionsIdentity includes all the base logic like users, userroles, their stores, validators, etc.
AspNetCoreIdentity includes additional logic like signing in (SignInManager), cookie authorization, etc.

It would be better to reference ExtensionsIdentity from this project, as we only require functionality that's part of this base.

I've tested building the project on my own machine after this change, and adding a reference to the local `MR.AspNet.Identity.EntityFramework6.dll` binary directly from my main project. Seems to work.

### CODE REVIEW

**Comment by mrahhal** [comment]
Thanks. Will review soon.

File: samples/Basic.Tests/TestHost.cs [hunk 1]
```diff
@@ -22,8 +22,8 @@ private IServiceProvider CreateProvider()
 		{
 			var services = new ServiceCollection();
 			services.AddLogging();
-			services
-				.AddIdentity<AppUser, AppRole>()
+			services.AddIdentityCore<AppUser>(_ => { })
+				.AddRoles<AppRole>()
 				.AddUserStore<InMemoryUserStore<AppUser, AppRole>>()
```
Comment (mrahhal) [inline samples/Basic.Tests/TestHost.cs]: Ok so. It's true that we don't need to depend on anything besides the Core package. But I guess apps will still probably use `.AddIdentity<AppUser, AppRole>()`, right? So the samples will stay the same. And because our main project now drops the "Microsoft.AspNetCore.Identity" dependency, we should just add it in this sample instead, and revert this change.

File: samples/Basic/Startup.cs [hunk 2]
```diff
@@ -12,7 +12,9 @@ public void ConfigureServices(IServiceCollection services)
 		{
 			services.AddLogging();
 			services.AddTransient((_) => new AppDbContext("Server=(localdb)\\mssqllocaldb;Database=MR.AspNet.Identity.EntityFramework6-Basic;Trusted_Connection=True;MultipleActiveResultSets=true"));
-			services.AddIdentity<AppUser, AppRole>().AddEntityFrameworkStores<AppDbContext>();
+			services.AddIdentityCore<AppUser>(_ => { })
+				.AddRoles<AppRole>()
+				.AddEntityFrameworkStores<AppDbContext>();
```
Comment (mrahhal) [inline samples/Basic/Startup.cs]: Same as above, just revert this change.

**Review by mrahhal — CHANGES_REQUESTED** [review]

File: samples/Basic.Tests/TestHost.cs (hunk 1 above)
Comment (nphmuller) [reply samples/Basic.Tests/TestHost.cs]: Agreed that the default sample should still be AddIdentity, as it still seems to be the main use case. I could add an extra sample for Core use, but since the code is pretty much the same I don't think it's worth it.

AddIdentityCore will probably be used for apps that only provide REST API's, and don't need the web page part of the MVC package. That's where I'm using it for at least.

Comment (mrahhal) [reply samples/Basic.Tests/TestHost.cs]: > AddIdentityCore will probably be used for apps that only provide REST API's

That's what I needed to know. Haven't yet had the time to go deep into how things changed in 2.0 😅 

Agreed. Another sample for usage with Core would be better, but it's also not that much needed. So feel free to do whatever is appropriate.

Comment (nphmuller) [reply samples/Basic.Tests/TestHost.cs]: Think I'll skip the additional sample then ;)
Changes are pushed.

**Comment by mrahhal** [comment]
Thanks a lot!
